buzzmobile/process/gps_mapper/interpolate.py

```diff
@@ -0,0 +1,151 @@
+              the window at in pixels
+    angle: angle in radians to rotate rectangular region by (counterclockwise)
+    height, width: dimensions of the output image in pixels
+    ----------------------------------------------------------------------------
+    """
+    parallel = (math.cos(angle), math.sin(angle))
+    perpendicular = (-math.sin(angle), math.cos(angle))
+    horizontal = location[0] - parallel[0] * (width/2) - perpendicular[0] * (height/1)
+    vertical = location[1] - parallel[1] * (width/2) - perpendicular[1] * (height/1)
+    rotation_matrix = np.array([[parallel[0], perpendicular[0], horizontal], [parallel[1], perpendicular[1], vertical]])
+    return cv2.warpAffine(image, rotation_matrix, (width, height), flags=cv2.WARP_INVERSE_MAP)
+
+def xwindow(points, location, angle, height=500, width=500):
+    """
+    Takes a list of points, a location, an angle in radians, and optionally a height and width
+    in order to return the angled rectangular region of the image of the specified size,
+    with the location specifying the bottom middle point of the image. The image itself is
+    the blurred interpolation of all the points (we only construct the relevant portion of the
+    rotated image).
+    ----------------------------------------------------------------------------
+    points: list of (x, y) tuples representing points
+    location: tuple of (x, y) representing the current location to center the bottom of
+              the window at in pixels
+    angle: angle in radians to rotate rectangular region by (counterclockwise)
+    height, width: dimensions of the output image in pixels
+    ----------------------------------------------------------------------------
+    """
+    out = []
+    parallel = (math.cos(angle), -math.sin(angle))
+    perpendicular = (math.sin(angle), math.cos(angle))
+    for (x, y) in points:
+        x_ = x * parallel[0] + y * parallel[1] + location[0] - location[0] * parallel[0] - location[1] * parallel[1]
+        y_ = x * perpendicular[0] + y * perpendicular[1] + location[1] - location[0] * perpendicular[0] - location[1] * perpendicular[1]
+        # when adding the points back, translate them to the proper location for the final image
+        out.append((x_ + (width/2 - location[0]), y_ + (height - location[1])))
+    return interpolate([(int(round(x)), int(round(y))) for (x, y) in out], 3, 3, width, height)
+
+def dimensions(points):
+    """
+    Takes a set of latitude and longitude points and returns the width and height in kilometers.
+    """
+    x_vals = [x for (x, y) in points]
+    y_vals = [y for (x, y) in points]
+    top_left = (min(x_vals), max(y_vals))
+    bottom_right = (max(x_vals), min(y_vals))
+    x_range = haversine(top_left[0], top_left[1], bottom_right[0], top_left[1])
+    y_range = haversine(top_left[0], top_left[1], top_left[0], bottom_right[1])
+    return y_range, x_range, top_left, bottom_right
+
+def main():
+    # this is just for testing
+    """
```

Comment:
Can you remove main, since it's just for testing and everything is commented out?